### ISSUE TITLE
It's pruning too much (dependency graph issue?)

### ISSUE THREAD

**Issue by skerit** [tomasklaen/postcss-prune-var#5]
Here's some beautiful HTML & SCSS for you:

```html
<div class="example">
	<div class="child">
		<div class="row">
			<div class="cell">
				1
			</div>
		</div>
		<div class="row">
			<div class="cell">
				2
			</div>
		</div>
		<div class="row">
			<div class="cell">
				3
			</div>
		</div>
		<div class="row">
			<div class="cell">
				4
			</div>
		</div>
	</div>
</div>
```

```scss
:root {
	--color-state-bg-striped: green;
}

.example {
	--color-component-bg: red;
	--color-state-bg: var(--color-component-bg);

	.cell {
		background-color: var(--color-state-bg);
	}

	.child > .row:nth-of-type(odd) > * {
		--color-state-bg: var(--color-state-bg-striped);
	}
}
```

Without pruning, this is the result in the browser:

![image](https://github.com/tomasklaen/postcss-prune-var/assets/755212/5145aa0f-66e6-42e0-b7a4-b8c13d6d1aa7)

However, with pruning, this is the generated CSS:

```css
:root {
}

.example {
    --color-component-bg: red;
    --color-state-bg: var(--color-component-bg);
}

.example .cell {
    background-color: var(--color-state-bg);
}

.example .child > .row:nth-of-type(odd) > * {
    --color-state-bg: var(--color-state-bg-striped);
}
```

So it removes the declaration in `:root`.
It always does this, even when the last rule is changed from `.example .child > .row:nth-of-type(odd) > *` to just `.example .child`,
the declaration in root is deleted.

**Comment by tomasklaen** [comment]
I published 1.1.2 with the fix for your example, but refer back if there are some more issues.

**Comment by skerit** [comment]
@tomasklaen Thanks, works great now!
